File: ApplicationForFixingWashingMachines/src/main/java/com/example/applicationforfixingwashingmachines/entity/Repair.java
package com.example.applicationforfixingwashingmachines.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.Where;

import java.time.LocalDate;

@Entity
@Getter
@Setter
@SQLDelete(sql = "UPDATE burmistrov.public.repair SET deleted = true WHERE id=?")
@Where(clause = "deleted=false")
public class Repair {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String phoneNumber;

    private String email;

    private double price;

    private String fullName;

    private LocalDate date;

    private boolean deleted = Boolean.FALSE;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "problems_id")
    private Problem problem;

}

File: frontend/src/app/models/Repair.ts
export interface Repair {
  id: number;
  phoneNumber: string;
  email: string;
  price: number | undefined;
  fullName: string;
  date: Date;
  problemId: number;
}

export class RepairImpl implements Repair{
  date: Date = new Date();
  email: string = "";
  fullName: string = "";
  id: number = -1;
  phoneNumber: string = "";
  price: undefined = undefined;
  problemId: number = -1;

}
